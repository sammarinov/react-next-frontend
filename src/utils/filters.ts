import { IFeatured, IArticle } from "@Redux/app/type";

export const filterFeatured = (featured: Array<IFeatured>, limit: number = -1) => {
    featured.sort((a, b) => {
        if (a.topFeatured && !b.topFeatured) return 1;
        if (!a.topFeatured && b.topFeatured) return -1;
        return parseInt(a.price) - parseInt(b.price);
    })

    if (limit > 0) return featured.slice(0, limit);
    return featured;
}

export const filterArticles = (articles: Array<IArticle>, limit: number = -1) => {
    articles.sort((a, b) => {
        return Date.parse(a.created) - Date.parse(b.created);
    })

    if (limit > 0) return articles.slice(0, limit);
    return articles;
}