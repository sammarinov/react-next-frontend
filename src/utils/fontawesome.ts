import { library } from '@fortawesome/fontawesome-svg-core';

import {
    faArrowRight
} from '@fortawesome/free-solid-svg-icons';

import {
    faTwitter,
    faFacebook,
    faInstagram,
    faYoutube
} from '@fortawesome/free-brands-svg-icons';

library.add(
    faArrowRight,
    faTwitter,
    faFacebook,
    faInstagram,
    faYoutube
)