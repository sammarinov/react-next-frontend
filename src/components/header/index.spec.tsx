import * as React from 'react';

import { render } from '@Test/utils';
import Header from '.';

describe('Header', () => {
  it('should match snapshot', () => {
    const { container } = render(<Header>Test</Header>);

    expect(container).toMatchSnapshot();
  });
});
