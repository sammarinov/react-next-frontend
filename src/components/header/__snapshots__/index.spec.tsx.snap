// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
.c1 {
  height: 70px;
}

.c0 {
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #102027;
  color: #ffffff;
  height: 80px;
}

<div>
  <header
    class="c0"
  >
    <img
      class="c1"
      src="/images/logo.gif"
    />
  </header>
</div>
`;
